File: bookstore-pos/frontend/src/api/audit.ts
import { api } from "./client";

export type AuditLog = {
  id: number;
  user_id: number | null;
  action: string;
  entity: string;
  entity_id: string;
  details: string;
  created_at: string;
};

export const listAuditLogs = async (): Promise<AuditLog[]> => {
  const res = await api.get("/audit");
  return res.data;
};

File: bookstore-pos/backend/app/models/audit.py
from datetime import datetime
from sqlalchemy import DateTime, ForeignKey, Integer, String
from sqlalchemy.orm import Mapped, mapped_column

from app.db.base import Base


class AuditLog(Base):
    __tablename__ = "audit_logs"

    id: Mapped[int] = mapped_column(Integer, primary_key=True)
    user_id: Mapped[int | None] = mapped_column(ForeignKey("users.id"), nullable=True)
    action: Mapped[str] = mapped_column(String(50))
    entity: Mapped[str] = mapped_column(String(50))
    entity_id: Mapped[str] = mapped_column(String(50))
    details: Mapped[str] = mapped_column(String(255), default="")
    created_at: Mapped[datetime] = mapped_column(DateTime, default=datetime.utcnow)
